lib/topological_inventory/providers/common/operations/source.rb

```diff
@@ -22,10 +25,13 @@ class Source
         attr_accessor :params, :request_context, :source_id, :account_number
 
         def initialize(params = {}, request_context = nil)
-          self.params          = params
-          self.request_context = request_context
-          self.source_id       = params['source_id']
-          self.account_number  = params['external_tenant']
+          self.params           = params
+          self.request_context  = request_context
+          self.source_id        = params['source_id']
+          self.account_number   = params['external_tenant']
+          self.messaging_host   = ENV['QUEUE_HOST']
```

Comment:
I suggest to use [Messaging client](https://github.com/RedHatInsights/topological_inventory-ansible_tower/blob/9eb20112332fda219d4faafc2272ad904d27641c/lib/topological_inventory/ansible_tower/messaging_client.rb#L19) and split the class to providers-common and inherited tower class. 

Then the messaging host and port can be defined through input args like [here](https://github.com/RedHatInsights/topological_inventory-ansible_tower/blob/9eb20112332fda219d4faafc2272ad904d27641c/bin/ansible_tower-operations#L60)

WIth this refactoring, use can make method
```
def messaging_client
   TopologicalInventory::Providers::Common::MessagingClient.default
end
```

(because of the `@@default` variable, this method will return instance of repository-specific client, if available ([here](https://github.com/RedHatInsights/topological_inventory-ansible_tower/blob/9eb20112332fda219d4faafc2272ad904d27641c/bin/ansible_tower-operations#L60)))

and cache client inside (see the `cache_with_timeout` comment)
and use messaging_client.client.queue_host